src/deploy/functions/build.ts

```diff
@@ -279,13 +279,15 @@ export async function resolveBackend(
   userEnvOpt: UserEnvsOpts,
   userEnvs: Record<string, string>,
   nonInteractive?: boolean,
+  isEmulator = false,
```

Comment:
Optional: this list of args is quite long, and it would be easy to transpose nonInteractive and isEmulator. Consider switching to a parameter object. ie:
```
export async function resolveBackend( args: {
  build: Build,
  firebaseConfig: FirebaseConfig,
  userEnvOpt: UserEnvsOpts,
  userEnvs: Record<string, string>,
  nonInteractive?: boolean,
  isEmulator?: boolean,
  })
  ```